securedrop_client/gui/widgets.py

```diff
@@ -659,13 +663,21 @@ def on_source_changed(self) -> None:
                 conversation_wrapper = self.source_conversations[source.uuid]
                 conversation_wrapper.conversation_view.update_conversation(source.collection)
             else:
-                conversation_wrapper = SourceConversationWrapper(source, self.controller)
+                conversation_wrapper = SourceConversationWrapper(
+                    source, self.controller, self._state
+                )
                 self.source_conversations[source.uuid] = conversation_wrapper
 
             self.set_conversation(conversation_wrapper)
             logger.debug(
                 "Set conversation to the selected source with uuid: {}".format(source.uuid)
             )
+
+            conversation_id = None
+            source = self.source_list.get_selected_source()
+            if source is not None:
+                conversation_id = source.uuid
+            self.source_selection_changed.emit(conversation_id)
```

Comment:
Why not connecting the signals? :thinking:

Reply:
@creviera I remember why I didn't "just" connect the signals here.

- `source_selection_changed` is emitted with the `conversation_id`, while the originating `source_list.itemSelectionChanged` has no parameters
- because of that, the conversation ID needs to be retrieved
- and that is more than can be done in a `lambda` (which the linter insists must be a one-liner)
- I had two options:
  - create a method for that (practically a slot)
  - do it in the slot that is already connected tot the `source_list.itemSelectionChanged` signal
- I chose the latter because I found it easier to reason about than having two slots connected to the same signal

What do you think?

Reply:
Actually, let's go one level down and do that conversion in `SourceList`.

Reply:
Done